web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -50,43 +48,23 @@ export class LayerDialogComponent implements OnDestroy {
   layer?: Layer;
   layerName!: string;
   projectId?: string;
-  activeProject$: Observable<Project>;
   subscription: Subscription = new Subscription();
   fieldTypes = FieldType;
   fields: List<Field>;
 
   constructor(
-    @Inject(MAT_DIALOG_DATA) data: { layerId: string },
+    @Inject(MAT_DIALOG_DATA) data: { layerId: string; project: Project },
     private dialogRef: MatDialogRef<LayerDialogComponent>,
-    private projectService: ProjectService,
     private dataStoreService: DataStoreService,
     private router: Router,
     private confirmationDialog: MatDialog
   ) {
     this.lang = 'en';
     // Disable closing on clicks outside of dialog.
     dialogRef.disableClose = true;
+    this.fields = List<Field>();
     this.layerId = data.layerId;
-    this.activeProject$ = this.projectService.getActiveProject$();
-    this.subscription.add(
-      this.activeProject$.subscribe(project => {
-        this.onProjectLoaded(project);
-      })
-    );
-    const fields = List<Field>();
-    const fieldId = this.dataStoreService.generateId();
-    this.fields = fields.push(
-      new Field(
-        fieldId,
-        1,
-        StringMap({
-          en: '',
-        }),
-        false,
-        0,
-        undefined
-      )
-    );
+    this.onProjectLoaded(data.project);
```

Comment:
This is a big of a misnomer within the context of this component now. Maybe something like `init()` would be more appropriate?